ga4gh/backend.py

```diff
@@ -384,6 +386,49 @@ def setMaxResponseLength(self, maxResponseLength):
         """
         self._maxResponseLength = maxResponseLength
 
+    # private methods used to implement readsGenerator
+
+    def _getReadGroup(self, request):
+        if len(request.readGroupIds) != 1:
+            if len(request.readGroupIds) == 0:
+                msg = "Read search requires a readGroup to be specified"
+            else:
+                msg = "Read search over multiple readGroups not supported"
+            raise exceptions.NotImplementedException(msg)
+        readGroupId = request.readGroupIds[0]
+        try:
+            readGroup = self._readGroupIdMap[request.readGroupIds[0]]
+        except KeyError:
+            raise exceptions.ReadGroupNotFoundException(readGroupId)
+        return readGroup
+
+    # private methods used to implement variantsGenerator
+
+    def _getVariantSet(self, request):
+        if len(request.variantSetIds) != 1:
+            if len(request.variantSetIds) == 0:
+                msg = "Variant search requires specifying a variantSet"
+            else:
+                msg = ("Variant search over multiple variantSets "
+                       "not supported")
+            raise exceptions.NotImplementedException(msg)
+        variantSetId = request.variantSetIds[0]
+        try:
+            variantSet = self._variantSetIdMap[variantSetId]
+        except KeyError:
+            raise exceptions.VariantSetNotFoundException(variantSetId)
+        return variantSet
+
+    # private methods common to various interval generators
+
+    def _getCounters(self, request):
```

Comment:
I'm not totally convinced by this name... `getIntervalCounters` or `getIntervalSearchToken` or something? `_getCounters` could mean almost anything.